src/types.ts

```diff
@@ -197,8 +197,43 @@ export interface ChatCompletionOptions {
    * https://platform.openai.com/docs/api-reference/chat/create#chat/create-user
    */
   user?: string;
+
+  /**
+   * A list of functions the model may generate JSON inputs for.
+   * https://platform.openai.com/docs/api-reference/chat/create#chat/create-functions
+   */
+    functions?: GPTFunction[];
 }
 
+type GPTFunction = {
+  name: string;
+  description: string;
+  parameters: ObjectSchema;
+};
+
+type JSONSchema = ObjectSchema | StringSchema | NumberSchema | BooleanSchema;
```

Comment:
These might not be the full set of supported types and some of the types might be missing optional properties... but it's a start

Reply:
Good start, especially given the lack of docs on this. This seems fine to me.

Reply:
Testing `Array` rn and seems to work...  The docs also mention `anyOf`, but I assume they also support `allOf`, `oneOf`.   Might just wait and add these in the future as we have time to test they actually work.